src/network/websockets/index.ts

```diff
@@ -38,6 +57,7 @@ export function makeWsProtocolAdapter(
 
       const handlers: { [action: string]: (msg: any) => void } = {
         ping: () => {},
```

Comment:
Correct me if I'm overlooking it, but shouldn't the `pong` message be sent back in response to a `ping` as part of this PR? Otherwise if this PR is merged the socket will reconnect every 15 seconds when the ping timeout expires.

From the spec:
>The server must immediately respond with a pong message

Reply:
This PR modifies the client-side code. Our spec makes the client send pings and the server respond with pongs, not the other way around. So this ping handler is not even necessary here.

restartPingMachine is called when receiving a pong, it’s called on any message from the server, in socket.onmessage.